File: api/app/Http/Controllers/Api/CashboxController.php
<?php
// app/Http/Controllers/Api/CashboxController.php

namespace App\Http\Controllers\Api;

use App\Http\Controllers\Controller;
use App\Models\Cashbox;
use App\Services\LedgerService;
use Illuminate\Http\Request;

class CashboxController extends Controller
{
    public function index()
    {
        $boxes = Cashbox::where('is_active', true)->orderBy('name')->get();

        return response()->json([
            'data'  => $boxes,
            'total' => $boxes->sum(fn ($b) => (float) $b->balance),
        ]);
    }

    public function store(Request $request)
    {
        $data = $request->validate([
            'name'     => ['required', 'string', 'max:255'],
            'type'     => ['required', 'in:cash,bank'],
            'balance'  => ['nullable', 'numeric'],
            'currency' => ['nullable', 'string', 'size:3'],
        ]);

        return response()->json(Cashbox::create($data)->fresh(), 201);
    }

    public function show(Request $request, Cashbox $cashbox)
    {
        return response()->json([
            'cashbox'      => $cashbox,
            'transactions' => $cashbox->transactions()
                ->orderByDesc('id')
                ->paginate(min((int) $request->integer('per_page', 25), 100)),
        ]);
    }

    public function update(Request $request, Cashbox $cashbox)
    {
        $cashbox->update($request->validate([
            'name'      => ['sometimes', 'string', 'max:255'],
            'is_active' => ['nullable', 'boolean'],
        ]));

        return $cashbox->fresh();
    }

    public function transfer(Request $request, LedgerService $ledger)
    {
        $data = $request->validate([
            'from_id' => ['required', 'integer', 'exists:cashboxes,id'],
            'to_id'   => ['required', 'integer', 'exists:cashboxes,id'],
            'amount'  => ['required', 'numeric', 'min:0.01'],
        ]);

        $result = $ledger->transfer($data['from_id'], $data['to_id'], (float) $data['amount'], $request->user()->id);

        return response()->json(['message' => 'Transfer tamamlandı.', ...$result]);
    }
}

File: api-net/Controllers/CashboxController.cs
using Liftonom.Api.Data;
using Liftonom.Api.Models;
using Liftonom.Api.Services;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;

namespace Liftonom.Api.Controllers;

[ApiController]
[Authorize]
[Route("api/v1/cashboxes")]
public class CashboxController(AppDbContext db, LedgerService ledger) : ControllerBase
{
    public record CashboxDto(string Name, string Type, decimal? Balance, string? Currency);
    public record TransferDto(long FromId, long ToId, decimal Amount);

    [HttpGet]
    public async Task<IActionResult> Index()
    {
        var boxes = await db.Cashboxes.Where(c => c.IsActive).OrderBy(c => c.Name).ToListAsync();
        return Ok(new { data = boxes, total = boxes.Sum(c => c.Balance) });
    }

    [HttpPost]
    public async Task<IActionResult> Store(CashboxDto dto)
    {
        var now = DateTime.UtcNow;
        var c = new Cashbox
        {
            TenantId = db.CurrentTenantId!.Value, Name = dto.Name, Type = dto.Type,
            Balance = dto.Balance ?? 0, Currency = dto.Currency ?? "TRY", CreatedAt = now, UpdatedAt = now,
        };
        db.Cashboxes.Add(c);
        await db.SaveChangesAsync();
        return StatusCode(201, c);
    }

    [HttpPost("transfer")]
    public async Task<IActionResult> Transfer(TransferDto dto)
    {
        var uid = long.Parse(User.FindFirst("uid")!.Value);
        var (from, to) = await ledger.TransferAsync(dto.FromId, dto.ToId, dto.Amount, uid);
        return Ok(new { message = "Transfer tamamlandı.", from_balance = from, to_balance = to });
    }

    [HttpGet("{id:long}")]
    public async Task<IActionResult> Show(long id, [FromQuery(Name = "per_page")] int perPage = 25, [FromQuery] int page = 1)
    {
        var c = await db.Cashboxes.FirstOrDefaultAsync(x => x.Id == id) ?? throw new ApiException(404, "Kasa bulunamadı.");
        var txns = await db.CashboxTransactions.Where(t => t.CashboxId == id).OrderByDescending(t => t.Id).ToPagedAsync(page, perPage);
        return Ok(new { cashbox = c, transactions = txns });
    }

    [HttpPut("{id:long}")]
    public async Task<IActionResult> Update(long id, [FromBody] CashboxDto dto)
    {
        var c = await db.Cashboxes.FirstOrDefaultAsync(x => x.Id == id) ?? throw new ApiException(404, "Kasa bulunamadı.");
        if (!string.IsNullOrWhiteSpace(dto.Name)) c.Name = dto.Name;
        c.UpdatedAt = DateTime.UtcNow;
        await db.SaveChangesAsync();
        return Ok(c);
    }
}
